include/turnstile.h

```diff
@@ -10,7 +10,8 @@ void turnstile_init(turnstile_t *);
 
 /* This puts currently running thread on the turnstile,
  * puts it in waiting state and yields. While waiting on turnstile
- * thread cannot be run. This function is done under critical section. */
+ * thread cannot be run. This function has to be called with critical section,
```

Comment:
> Sorry to disappoint you about not asking ;)

No problem ;-)

Yeah. You're right – it's even explicitely mentioned in [the comment](http://bxr.su/FreeBSD/sys/kern/subr_turnstile.c#674). Frankly, I cannot justify the reason behind doing so. IMO a reader will get confused when they don't see `cs_enter` / `cs_leave` symmetry. Let's hold on consistency as much as we can :)